programs/ssl/ssl_client2.c

```diff
@@ -1718,7 +1718,8 @@ int main( int argc, char *argv[] )
         }
         else if( strcmp( p, "query_config" ) == 0 )
         {
-            return query_config( q );
+            ret = query_config( q );
+            goto exit;
```

Comment:
This changes the behaviour:
- if `MBEDTLS_ERROR_C` is defined and `query_config( q )` == 1, then an extra error message will be printed in the log (line 4278)
- if `MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED` is defined, then the return value will be overridden (line 4322)
This is probably the reason why the CI is failing.

Reply:
I still think that this changes behaviour. I just noticed that I gave the line numbers for `ssl_server2.c` in my previous comment and that you fixed it there, but the same problem holds here too.